src/app/index.tsx

```diff
@@ -20,7 +21,19 @@ const App: React.FunctionComponent = () => (
     <PollingContextProvider>
       <LocalStorageContextProvider>
         <NetworkContextProvider>
-          <Router>
+          <Router
+            getUserConfirmation={(message: string, callback: (ok: boolean) => void) => {
+              const isAllowed = message === RouteGuardOptions.permit || message === '';
+              if (
+                message !== RouteGuardOptions.prevent &&
+                message !== RouteGuardOptions.permit &&
+                message !== ''
+              ) {
+                confirm(message) ? callback(true) : callback(false);
```

Comment:
Ah, is this logic just here so that we could use the regular <Prompt> if we really want to? If we make RouteGuard into a reusable component in lib-ui or PF, does the consumer need to make sure they put this code in their router as well?

Reply:
Yes exactly - it could be cleaned up if we don't care to have support for `Prompt` from react-router-dom (I don't know why we'd ever use the old version given it doesn't automatically prevent browser tab close like you mentioned). I'd personally prefer to remove that old support and just require devs to always use the newer version, but I'm happy to implement whatever everyone thinks is best.

It does look like any consumer will have to modify this `getUserConfirmation` prop in some way in order to customize the prompt behavior. If I can find a way to avoid it I surely will, but as of this moment it looks like some type of modification will be required.